src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -184,7 +187,7 @@ public boolean topicExists(final String topic) throws TopicRepositoryException {
             final ProducerRecord<String, String> kafkaRecord = new ProducerRecord<>(
                     topicId,
                     KafkaCursor.toKafkaPartition(item.getPartition()),
-                    item.getPartition(),
+                    item.getEventKey(),
```

Comment:
Do you know why we used to publish the partition as a key? Is it somehow change anything?

Reply:
I think we were using partition as a key accidentally. In general if you directly specify partition to publish to - it will just ignore the key, it is stated like that in documentation. So I don't expect anything to be changed.
Or do you suggest to keep using partition as a key for cleanup_policy=delete? (not to break anything)

Reply:
I think we can test it, no problem. Good that you already checked the docs and they say it should not change anything. We just need to confirm 👍